### PR TITLE
Igal/controller logs

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1925,22 +1929,40 @@ func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, para
...
 }
 
-func (b *bareMetalInventory) getLogFileForDownload(ctx context.Context, clusterId *strfmt.UUID, hostId *strfmt.UUID) (string, error) {
+func (b *bareMetalInventory) getLogFileForDownload(ctx context.Context, clusterId *strfmt.UUID, hostId *strfmt.UUID, logsType string) (string, string, error) {
 	var fileName string
-	if hostId != nil {
-		host, err := b.getHost(ctx, clusterId.String(), hostId.String())
-		if err != nil {
-			return "", err
+	var donwloadFileName string
```
Comment (avishayt) [inline internal/bminventory/inventory.go:1934]:
```suggestion
	var downloadFileName string
```

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -1925,22 +1929,40 @@ func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, para
...
+	c, err := b.getCluster(ctx, clusterId.String())
+	if err != nil {
+		return "", "", err
+	}
+	switch logsType {
+	case string(models.LogsTypeHost):
+		if hostId == nil {
+			return "", "", common.NewApiError(http.StatusBadRequest, errors.Errorf("Host id must be provided"))
```
Comment (avishayt) [inline internal/bminventory/inventory.go:1942]:
```suggestion
			return "", "", common.NewApiError(http.StatusBadRequest, errors.Errorf("Host ID must be provided"))
```

**Review by avishayt — APPROVED** [review]

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2515,16 +2537,40 @@ func (b *bareMetalInventory) UploadHostLogs(ctx context.Context, params installe
 			log.WithError(err).Warnf("Failed to delete temporary files used for upload")
 		}
 	}()
+	if params.LogsType == string(models.LogsTypeHost) {
```
Comment (eranco74) [inline internal/bminventory/inventory.go:2540]: Add logging

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -2515,16 +2537,40 @@ func (b *bareMetalInventory) UploadHostLogs(ctx context.Context, params installe
...
+		err := b.uploadHostLogs(ctx, params.ClusterID.String(), params.HostID.String(), params.Upfile)
+		if err != nil {
+			return common.GenerateErrorResponder(err)
+		}
+		return installer.NewUploadLogsNoContent()
+	}
+	_, err := b.getCluster(ctx, params.ClusterID.String())
```
Comment (eranco74) [inline internal/bminventory/inventory.go:2547]: Add logging

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -2515,16 +2537,40 @@ func (b *bareMetalInventory) UploadHostLogs(ctx context.Context, params installe
...
+	if err != nil {
+		common.GenerateErrorResponder(err)
+	}
+	fileName := b.getLogsFullName(params.ClusterID.String(), params.LogsType)
+	log.Debugf("Start upload log file %s to bucket %s", fileName, b.S3Bucket)
+	err = b.objectHandler.UploadStream(ctx, params.Upfile, fileName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to upload %s to s3", fileName)
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
```
Comment (eranco74) [inline internal/bminventory/inventory.go:2557]: We set nothing on the cluster about these logs (like you do for the host logs...)

File: internal/bminventory/inventory.go [hunk 6]
```diff
@@ -1925,22 +1929,40 @@ func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, para
...
-			return "", err
+			return "", "", err
+		}
+		if hostObject.LogsCollectedAt == strfmt.DateTime(time.Time{}) {
+			return "", "", common.NewApiError(http.StatusNotFound, errors.Errorf("Logs for host %s were not found", hostId))
+		}
+		fileName = b.getLogsFullName(clusterId.String(), hostObject.ID.String())
+		donwloadFileName = fmt.Sprintf("%s_%s", hostutil.GetHostnameForMsg(hostObject), filepath.Base(fileName))
+	case string(models.LogsTypeController):
+		fileName = b.getLogsFullName(clusterId.String(), logsType)
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/bminventory/inventory.go:1955]: nope, here it is not relevant

File: internal/bminventory/inventory.go [hunk 7]
```diff
@@ -1925,22 +1929,40 @@ func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, para
...
+		donwloadFileName = fmt.Sprintf("%s_%s.tar.gz", clusterId, logsType)
+	default:
+		fileName, err = b.prepareClusterLogs(ctx, c)
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/bminventory/inventory.go:1958]: nope

File: internal/bminventory/inventory.go (hunk 4 above)
Comment (tsorya) [reply internal/bminventory/inventory.go:2547]: there is log inside getCluster

**Pull request merged**